include/eosio/to_json.hpp

```diff
@@ -167,6 +168,29 @@ void to_json(const std::vector<T>& obj, S& stream) {
    stream.write(']');
 }
 
+template <typename K, typename V, typename S>
+void to_json(const std::map<K, V>& obj, S& stream) {
```

Comment:
This is incompatible with ABI-conversion of structs defined in CDT which contain maps. It can throw of wasm-abi pretty badly.

Reply:
The root of the problem lies `abi_def`:
```
might_not_exist<std::map<eosio::name, kv_table_entry_def>> kv_tables{};
```

This breaks with existing convention since `map` isn't a built-in abi type.

Reply:
It will also throw off eosjs, which expects the json format of ABI to be described in ABI.

Reply:
See https://github.com/EOSIO/eosjs/blob/1cf9bf53d21d3f461f61970d3295ee1fc2ba56d1/src/abi.abi.json

Reply:
> This breaks with existing convention
That was intentional, since the current convention/format is somewhat obtuse/non standard JSON.

> which expects the json format of ABI to be described in ABI
Can we not update the ABI ABI to handle this?

Reply:
That'd require adding a new feature, expanding scope of the next release.

Reply:
That type of new feature (maps) requires serious additions to:
* abi_serializer (extreme care needed to avoid adding vulnerabilities)
* cdt (abi generation). Bad part: the ABI generated for maps and the JSON format will change drastically, breaking existing end users
* abieos
* eosjs